server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailQuerySerializer.scala

```diff
@@ -81,7 +81,11 @@ class EmailQuerySerializer @Inject()(mailboxIdFactory: MailboxId.Factory) {
     case _ => JsError(s"Expecting a JsString to represent a known operator")
   }
 
-  private implicit val filterConditionReads: Reads[FilterCondition] = Json.reads[FilterCondition]
+  private implicit val filterConditionReads: Reads[FilterCondition] = {
+    case JsObject(underlying) if underlying.keySet.diff(FilterCondition.SUPPORTED).nonEmpty => JsError(s"Unsupported filter")
```

Comment:
Can we say which filters were unsupported?

`JsError(s"Unsupported filter ${underlying.keySet.diff(FilterCondition.SUPPORTED).toSting}")`